source/common/http/request_id_extension_impl.cc

```diff
@@ -7,6 +7,20 @@
 namespace Envoy {
 namespace Http {
 
+namespace {
+
+// NoopRequestIDExtension is the implementation used outside of HTTP context.
+class NoopRequestIDExtension : public RequestIDExtension {
```

Comment:
nit: it would be nice to get coverage of the other methods here in the unit test you added.